ProfessorAssessor/Networking/Networker.m

```diff
@@ -58,9 +58,64 @@ + (void)buildReview:(Professor *)professor
         if (error == nil) {
             PFRelation *reviews = [course relationForKey:@"reviews"];
             [reviews addObject:newReview];
-            [course saveInBackground];
+            [course saveInBackgroundWithBlock:completion];
         }
     }];
 }
 
++ (void)fetchSchoolAndProfessorsWithCompletion:(
+                                                void(^)
+                                                (PFObject *_Nullable,
+                                                 NSError *_Nullable))completion {
+    PFQuery *query = [School query];
+
+    [query includeKeys:@[@"professors",
+                         @"professors.courses",
+                         @"professors.courses.reviews"]];
+
+    [query
+     getObjectInBackgroundWithId:[User currentUser].school.objectId
+     block:completion];
+}
+
++ (void)fetchReviewsForProfessor:(Professor *)professor
+                      forCourses:(NSArray<Course *> *)courses
+                  withCompletion:(
+                                  void(^)
+                                  (NSArray<Review *> *_Nullable objects,
+                                   NSError *_Nullable error))completion {
+    PFQuery *query = [Review query];
+
+    [query orderByDescending:@"createdAt"];
+    [query whereKey:@"professor" equalTo:professor];
+    [query whereKey:@"course" containedIn:courses];
+
+    [query findObjectsInBackgroundWithBlock:completion];
+}
+
+
++ (Course *)courseFromObject:(PFObject *)object
+              withCompletion:(
+                              void(^)(NSArray<Review *> *_Nullable objects,
+                                      NSError *_Nullable error))completion {
+    if (object) {
+        Course *course = [Course new];
+        course.identifier = object[@"objectId"];
+        course.createdAt = object[@"createdAt"];
+        course.updatedAt = object[@"updatedAt"];
+        course.name = object[@"name"];
+
+        PFQuery *reviewQuery = [Review query];
+
+        [reviewQuery orderByDescending:@"createdAt"];
+        [reviewQuery includeKey:@"course"];
+        [reviewQuery whereKey:@"course" equalTo:object];
+
+        [reviewQuery findObjectsInBackgroundWithBlock:completion];
+
+        return course;
+    }
+    return [Course new];
+}
```

Comment:
How come course is here but the others are not? I actually think this is the right place to put this but there's a detail that's wrong. I would do it more like this:

```
+ (Course *)courseFromObject:(PFObject *)object
              withCompletion:(
                              void(^)(NSArray<Review *> *_Nullable objects,
                                      NSError *_Nullable error))completion {
    if (object) {
        NSString *identifier = object[@"objectId"];
        NSDate *createdAt = object[@"createdAt"];
        /// etc...

        if (identifier && createdAt && otherRequiredThing) {
            return [Course courseWithIdentifier:identifier createdAt:createdAt etc...
        } else {
            return nil;
        }
    } else {
        return nil;
    }
}
``